File: SocialChallenge/NavigationController/JSCRootNavigationController.m
//
//  JSCRootNavigationController.m
//  SocialChallenge
//
//  Created by Javier Laguna on 09/08/2015.
//
//

#import "JSCRootNavigationController.h"

#import "JSCHomeViewController.h"

@interface JSCRootNavigationController () <UINavigationControllerDelegate>

/**
 Landing view controller.
 */
@property (nonatomic, strong) JSCHomeViewController *homeViewController;

@end

@implementation JSCRootNavigationController

#pragma mark - Init

- (instancetype)init
{
    self = [super initWithRootViewController:self.homeViewController];
    
    if (self)
    {
        self.delegate = self;
    }
    
    return self;
}

#pragma mark - Getters

- (JSCHomeViewController *)homeViewController
{
    if (!_homeViewController)
    {
        _homeViewController = [[JSCHomeViewController alloc] init];
    }
    
    return _homeViewController;
}

@end

File: SocialChallenge/NavigationController/JSCRootNavigationController.swift
//
//  JSCRootNavigationController.swift
//  SocialChallenge
//
//  Created by Javier Laguna on 15/03/2016.
//
//

import Foundation
import UIKit

/**
 Handles the navigation in the app.
*/
class JSCRootNavigationController: UINavigationController, UINavigationControllerDelegate {

    var homeViewController: JSCHomeViewController?
    
    //MARK: Init
    
    required init?(coder aDecoder: NSCoder) {
        
        fatalError("init(coder:) has not been implemented")
    }
    
    convenience init() {
        
        self.init(rootViewController: JSCHomeViewController.init())
        
        self.delegate = self
    }
    
    override init(rootViewController: UIViewController) {
        
        super.init(rootViewController: rootViewController)
        
        self.homeViewController = rootViewController as? JSCHomeViewController;
    }
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    }
}
